src/app/table/row/row.spec.ts

```
@@ -0,0 +1,116 @@
+            sheetState.getId() + ' b ' + numTimingFiles;
+        const valueDefinition = (sheetState: SheetState, column: Column, poiName?: string) =>
+            column.getValue(sheetState.getSheetTimingStates()[0], poiName);
+        const displayDefinition = (numTimingFiles: number, sheetState: SheetState, isSelectedRow: boolean) =>
+            isSelectedRow && numTimingFiles > sheetState.getId();
+        const defaultEnabled = false;
+
+        const row = new Row(id, dropdownText, displayIdDefinition, valueDefinition, displayDefinition, defaultEnabled);
+
+        expect(row.getId()).toBe(id);
+        expect(row.isEnabled()).toBe(defaultEnabled);
+
+        const dummySheetTimingState = new SheetTimingState(0, 0);
+        const dummySheetState = new SheetState(5, 13, 74, [dummySheetTimingState]);
+        const dummyPoiName = 'aPoiBDE';
+        const dummyNumTimingFiles = 1;
+        expect(row.getDisplayId(dummySheetState, dummyNumTimingFiles)).toBe(
+            displayIdDefinition(dummySheetState, dummyNumTimingFiles)
+        );
+        const dummyColumn = new Column(
+            6,
+            'DTextds',
+            (poiName: string) => 'headerText ' + poiName + 'randdddddom',
+            (poiName: string) => 'hoverText ' + poiName + 'randomnesss',
+            (sheetTimingState: SheetTimingState, poiName: string) =>
+                new Entry(sheetTimingState.getTimingId() + ' ABC ' + poiName),
+            false
+        );
+        expect(row.getValue(dummySheetState, dummyColumn, dummyPoiName).toString)
+            .toBe(valueDefinition(dummySheetState, dummyColumn, dummyPoiName).toString);
+        const dummyIsSelectedRow = false;
+        expect(row.canDisplay(dummyNumTimingFiles, dummySheetState, dummyIsSelectedRow))
+            .toBe(displayDefinition(dummyNumTimingFiles, dummySheetState, dummyIsSelectedRow));
+    });
+
+    it('should set and get enabled correctly', () => {
```

`'should throw errors when passing undefined parameters in construction'`?

---

Should be resolved now